README.md

```diff
@@ -60,20 +60,93 @@ Note: The only requirement for the ROCK is to have a `scripts/start-worker.sh`
 file, which will be used as the entry point for the charm to start the workload
 container.
 
-### Adding Environment Variables
+### Adding Secrets & Environment Variables
 
-The Temporal Worker operator can be used to inject environment variables that
-can be ingested by your workflows. This can be done using the Juju command line
-as follows:
+The Charmed Temporal Worker allows the user to configure multiple sources of
+environment variables and secrets to be injected into the workload container and
+consumed by the user's workflow definitions. These sources can be configured
+through the `secrets` config parameter of the charm. Below are the three sources
+of environment variables and secrets currently supported. A user may choose to
+use one, all or none of them. Once the `secrets.yaml` file is ready, it can be
+configured into the charm as follows:
 
 ```bash
-juju attach-resource temporal-worker-k8s env-file=path/to/.env
+juju config temporal-worker-k8s secrets=@/path/to/secrets.yaml
 ```
 
-#### **`.env`**
+These secrets can then be injested by the workflows by using the `os` package as
+follows:
 
+```python
+import os
+value1 = os.getenv("key1")
 ```
-VALUE=123
+
+#### Direct Environment Variables
+
+These are usually values that are not secret and can be stored as plaintext. An
+example is setting the application environment to `staging` or `production`.
+They can be set as follows:
+
+##### **`secrets.yaml`**
```

Comment:
what do you think about naming this `environment.yaml` and setting `env` as root (since all secrets are env but not vice versa)
a low impact change would be:
```
env:
    plain (or default):
        - key1: value1
    juju:
       ....
    vault:
        ...    
```
we could opt for a k8s-ish syntax too (more impact):
```
env:
    - name: key1
      value: value1
    - name: key2
      valueFromSecret:
          type: juju
          id: secret-id
          key: key2
    - name: key3
      valueFromSecret:
          type: vault
          id: secret-path
          key: key3    
```
or a more compact form:
```
env:
    - name: key1
      value: value1
    - name: key2
      valueFromSecret: juju@secret-id:key2
    - name: key3
      valueFromSecret: vault@secret-path:key3    
```